### ISSUE TITLE
Improve "Please specify by known `name=uuid`" error message

### ISSUE THREAD

**Issue by marius311** [JuliaLang/Pkg.jl#511]
I'm getting the following error message:

```
julia> Pkg.resolve()
 Resolving package versions...
ERROR: The following package names could not be resolved:
 * TreeViews=0cdf7f9c-6f0a-11e8-33c6-7b5c10fffe5f (not found in project, manifest or registry)
Please specify by known `name=uuid`.
```

My exact problem I don't think is too relevant to this Issue (I think the source of this is that a branch of Atom.jl that I'm using depends on an unregistered TreeViews, which I've added directly via `add github.com/.../TreeViews.jl`, but then something broke) 

My main suggestion is that something better than `Please specify by known name=uuid` should be printed. As a user, I have 1) no idea who is supposed to specify `name=uuid`, is it me or some package maintainer? and 2) if it is me, I have no idea where I'm supposed to do this, in some command I typed, or in a file somewhere? These are real questions I have, not being yet that familiar with Pkg3; it seems a slightly better message here could be really helpful.

**Comment by KristofferC** [comment]
Yeah, I think the original error message was designed to show up only when you did

```
add Foo
```

which then had the solution to instead do

```
add Foo=abcde...
```

but it should indeed be improved for this type of situations.

**Comment by marius311** [comment]
Ah I see, yea, in that case it would have been more obvious.
